pep-0589.rst

```diff
@@ -0,0 +1,645 @@
+
+    movie: Movie = {'name': 'Blade Runner',
+                    'year': 1982}
+
+An explicit ``Movie`` type annotation is generally needed, as
+otherwise an ordinary dictionary type could be assumed by a type
+checker, for backwards compatibility.  When a type checker can infer
+that a constructed dictionary object should be a TypedDict, an
+explicit annotation can be omitted.  A typical example is a dictionary
+object as a function argument.  In this example, a type checker is
+expected to infer that the dictionary argument should be understood as
+a TypedDict::
+
+    def record_movie(movie: Movie) -> None: ...
+
+    record_movie({'name': 'Blade Runner', 'year': 1982})
+
+Another example where a type checker should treat a dictionary display
+as a TypedDict is in an assignment to a variable with a previously
+declared TypedDict type::
+
+    movie: Movie
+    ...
+    movie = {'name': 'Blade Runner', 'year': 1982}
+
+Operations on ``movie`` can be checked by a static type checker::
+
+    movie['director'] = 'Ridley Scott'  # Error: invalid key 'director'
+    movie['year'] = '1982'  # Error: invalid value type ("int" expected)
+
+The code below should be rejected, since ``'title'`` is not a valid
+key, and the ``'name'`` key is missing::
+
+    movie2: Movie = {'title': 'Blade Runner',
+                     'year': 1982}
+
+The created TypedDict type object is not a real class object.  These
+are the only uses of the type a type checker is expected to allow:
+
+* It can be used in type annotations and in any context where an
+  arbitrary type hint is valid, such as in type aliases and as the
+  target type of a cast.
+
+* It can be used as a callable object with keyword arguments
+  corresponding to the TypedDict items.  Non-keyword arguments are not
+  allowed.  Example::
+
+      m = Movie(name='Blade Runner', year=1982)
+
+  When called, the TypedDict type object returns an ordinary
+  dictionary object at runtime::
+
+      print(type(m))  # <class 'dict'>
+
+* It can be used as a base class, but only when defining a derived
+  TypedDict.  This is discussed in more detail below.
+
+In particular, TypedDict type objects cannot be used in
+``isinstance()`` tests such as ``isinstance(d, Movie)``. The reason is
+that there is no existing support for checking types of dictionary
+item values, since ``isinstance()`` does not work with many PEP 484
+types, including common ones like ``List[str]``.  This would be needed
+for cases like this::
+
+    class Strings(TypedDict):
+        items: List[str]
+
+    print(isinstance({'items': [1]}))    # Should be False
+    print(isinstance({'items': ['x']}))  # Should be True
+
+The above use case is not supported.  This is consistent with how
+``isinstance()`` is not supported for ``List[str]``.
+
+
+Inheritance
+-----------
+
+It is possible to inherit from one or more TypedDict types using the
+class-based syntax.  A single class cannot inherit from both a
```

Comment:
The second sentence of the paragraph feels a bit out of place; it sounds like the example is going to be of inheritance from a TypedDict and a non-TypedDict.